diff --git a/dist/b-tree.js b/dist/b-tree.js
index 3d9b8251c755af1517a7ff091f41b2b4e1bccdb1..7bd44c2bb3fbdf6fb7fcb8cc7b7b9463cd2fb2dc 100644
--- a/dist/b-tree.js
+++ b/dist/b-tree.js
@@ -702,6 +702,9 @@ export class BTree {
         for (let i = segments.length - 1; i >= 0; --i) {
             const seg = segments[i];
             if (seg.node.tree === this) {
+                if (prior) {
+                    seg.node.nodes[seg.index] = prior;
+                }
                 return;
             }
             const newBranch = seg.node.clone(this);
@@ -720,7 +723,11 @@ export class BTree {
     }
 }
 function leafSibPath(path, sib, delta) {
-    return new Path(path.branches.map(b => b.clone()), sib, path.leafIndex + delta, path.on, path.version);
+    const branches = path.branches.map(b => b.clone());
+    if (branches.length > 0) {
+        branches[branches.length - 1].index += delta;
+    }
+    return new Path(branches, sib, path.leafIndex + delta, path.on, path.version);
 }
 class Split {
     key;
